integration_tests/scheduled_cork_test.go

```diff
@@ -129,82 +131,88 @@ func (s *IntegrationTestSuite) TestScheduledCork() {
-			if err != nil {
-				s.T().Logf("error: %s", err)
-				return false
-			}
-
 			node, err := clientCtx.GetNode()
 			s.Require().NoError(err)
 			status, err := node.Status(context.Background())
 			s.Require().NoError(err)
-			blockHeight := status.SyncInfo.LatestBlockHeight
 
-			gbClient := gbtypes.NewQueryClient(clientCtx)
-			gbRes, err := gbClient.ContractCallTxs(context.Background(), &gbtypes.ContractCallTxsRequest{
-				Pagination: nil,
-			})
-			s.Require().NoError(err)
+			currentHeight := status.SyncInfo.LatestBlockHeight
+			if currentHeight > (targetBlockHeight + 1) {
+				blockHeight = uint64(status.SyncInfo.LatestBlockHeight)
+				return true
+			} else if currentHeight < targetBlockHeight {
+				res, err := corkQueryClient.QueryScheduledCorks(context.Background(), &types.QueryScheduledCorksRequest{})
+				if err != nil {
+					s.T().Logf("error: %s", err)
+					return false
+				}
 
-			if blockHeight < (targetBlockHeight - 2) {
-				// verify that tbe scheduled cork has not yet been consumed, and that the counter has not been incremented
+				// verify that the scheduled corks have not yet been consumed
 				s.Require().Len(res.Corks, len(s.chain.validators))
-				s.Require().Equal(counterContract, common.HexToAddress(res.Corks[0].Cork.TargetContractAddress))
-				s.Require().Len(gbRes.Calls, 0)
-			} else if blockHeight > (targetBlockHeight + 1) {
-				// verify that block height has been passed, cork consumed, contractcalltx created
-				s.Require().Len(res.Corks, 0)
-				s.Require().Len(gbRes.Calls, 1)
-
-				// this is the only situation where this loop will complete
-				return true
 			}
 
 			return false
-		}, 3*time.Minute, 1*time.Second, "count was never updated")
-
-		s.T().Logf("verify count was updated")
+		}, 3*time.Minute, 1*time.Second, "never reached scheduled height")
+
+		s.T().Log("verify the cork was approved")
+		resultRes, err := corkQueryClient.QueryCorkResult(context.Background(), &types.QueryCorkResultRequest{Id: corkID})
+		s.Require().NoError(err, "failed to query cork result")
+		s.Require().True(resultRes.CorkResult.Approved, "cork was not approved")
+		s.Require().Equal(counterContract, common.HexToAddress(resultRes.CorkResult.Cork.TargetContractAddress))
+
+		s.T().Log("verify scheduled corks were deleted")
+		res, err := corkQueryClient.QueryScheduledCorksByBlockHeight(context.Background(), &types.QueryScheduledCorksByBlockHeightRequest{BlockHeight: blockHeight})
+		s.Require().NoError(err, "failed to query scheduled corks by height")
+		s.Require().Len(res.Corks, 0)
+
+		s.T().Log("verify gravity contract call tx was created")
+		gbRes, err := gbClient.ContractCallTxs(context.Background(), &gbtypes.ContractCallTxsRequest{
+			Pagination: nil,
+		})
+		s.Require().NoError(err, "failed to query gravity contract call txs")
+		s.Require().Len(gbRes.Calls, 1)
+
+		s.T().Log("verify count was updated")
```

Comment:
We should add a test for the governance version of a scheduled cork into this integration test as well.